src/transformation.rs

```diff
@@ -17,19 +16,26 @@ pub fn transform_relative_pt(
     (transformed_point.x, transformed_point.y)
 }
 
-pub fn transform_relative_point(
-    tf: &rosrust_msg::geometry_msgs::Transform,
-    pt: (f64, f64),
-) -> (f64, f64) {
-    let tra = Translation3::new(tf.translation.x, tf.translation.y, tf.translation.z);
-    // 3: z 2: y, 1: x, 0: w
+pub fn ros_to_minimal(tf: &rosrust_msg::geometry_msgs::Transform) -> (f64, f64, f64) {
     let rot = UnitQuaternion::new_normalize(Quaternion::new(
         tf.rotation.w,
         tf.rotation.x,
         tf.rotation.y,
         tf.rotation.z,
     ));
-    let isometry = Isometry3::from_parts(tra, rot);
-    let transformed_point = isometry.transform_point(&Point3::new(pt.0, pt.1, 0.));
-    (transformed_point.x, transformed_point.y)
+    let (_roll, _pitch, yaw) = rot.euler_angles();
+    (tf.translation.x, tf.translation.y, yaw)
+}
+
+pub fn minimal_to_ros(x: f64, y: f64, yaw: f64) -> rosrust_msg::geometry_msgs::Transform {
```

Comment:
not a huge fan of 'minimal'. it doesn't feel generic enough to justify it in this module. I thought about offering this with optional parameters for xyz and rpm ( or qx, qz..).

but this will be tricky[
](https://vidify.org/blog/rust-parameters/)

Reply:
https://vidify.org/blog/rust-parameters/

Reply:
The bane of not having transformations.py
I also dislike the name, something like get_position_vector_from_pose or extract vector
However, (not part of this PR) but maybe it makes sense to create a QOL class to handle this in another crate?

Reply:
we are working in 2D here: the minimal representation of a pose is x, y, angle. The only more "generic" thing is a 3D representation. Is anybody ok with a struct like "Pose2D" (or propose a similar name)?

Reply:
Pose2D is fine with me. Nalgebra also has 2d "poses": https://docs.rs/nalgebra/0.10.1/nalgebra/struct.Isometry2.html